wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/convert/ProjectConverter.java

```diff
@@ -186,5 +190,9 @@ private void updateSourceStorage(Source devfileSource, SourceStorageImpl sourceS
     } else if (commitId != null) {
       sourceStorage.getParameters().put(COMMIT_ID_PARAMETER_NAME, commitId);
     }
+
+    if (devfileSource.getSparseCheckoutDir() != null) {
+      sourceStorage.getParameters().put("keepDir", devfileSource.getSparseCheckoutDir());
```

Comment:
why not `SPARSE_CHECKOUT_DIR_PARAMETER_NAME` ?

Reply:
Because it was decided to use `sparseCheckoutDir` for Che 7 (which is SPARSE_CHECKOUT_DIR_PARAMETER_NAME), whereas `keepDir` is from Che 6.
I didn't add old Che 6 key into interface to avoid confusion.

Reply:
ok thanks